cloud/aws/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,131 @@
+// Copyright © 2018 The Kubernetes Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrumentation
+
+import (
+	"bytes"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/glog"
+	"go.opencensus.io/plugin/ochttp"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"go.opencensus.io/trace"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptrace"
+	"net/url"
+	"strings"
+)
+
+var (
+	AWSOperation, _ = tag.NewKey("aws.operation")
+	AWSService, _   = tag.NewKey("aws.service")
+	AWSRegion, _    = tag.NewKey("aws.region")
+)
+
+func awsRequestAction(r *http.Request) string {
```

Comment:
this reads the body, parses the query params, then I'm pretty confused on line 50, why does this put the body back into a reader that isn't closable? it then returns the query value for the Action paramter

Reply:
Just experimenting at the moment. Was hacking around to get the body without destroying the request. It doesn't yet actually do the thing I want it to.

Reply:
ohh ok, didn't realize this was experimental code. hack on!